stdlib/public/core/ThreadLocalStorage.swift

```diff
@@ -116,10 +116,10 @@ internal func _destroyTLS(_ ptr: UnsafeMutableRawPointer?) {
 }
 
 // Lazily created global key for use with pthread TLS
-internal let _tlsKey: __swift_pthread_key_t = {
-  let sentinelValue = __swift_pthread_key_t.max
-  var key: __swift_pthread_key_t = sentinelValue
-  let success = _swift_stdlib_pthread_key_create(&key, _destroyTLS)
+internal let _tlsKey: __swift_thread_key_t = {
+  let sentinelValue = __swift_thread_key_t.max
+  var key: __swift_thread_key_t = sentinelValue
+  let success = _swift_stdlib_thread_key_create(&key, _destroyTLS)
```

Comment:
Is there a way to change the calling convention for `_destroyTLS`?  On Windows (x86), this should be a `__stdcall` function.

Reply:
I don't think the Swift compiler can manipulate Windows calling conventions yet. It should be straightforward to rewrite _destroyTLS in C, or write a C __stdcall function that calls back into Swift.

Reply:
We could introduce a `@_callingconv` attribute.  But, it sounds like it would be easier to move it to swift-shims and rewrite it in C.

Reply:
A shim that calls back into the stdlib for deinitialization would be my preferred approach. We have non-trivial deinitialization logic that ideally would be kept in Swift. We likewise probably want allocate/deallocate parity, and Swift knows the sizes of the TLS buffer already.